content/training/R_Basics/Advanced_Functions.Rmd

```diff
@@ -12,82 +12,99 @@ weight: 12
 library(sitrep)
 library(here)
 library(dplyr)
+library(forcats)
 
-linelist_raw <- gen_data("AJS")
-linelist_cleaned <- linelist_raw
+linelist_cleaned <- gen_data("AJS")
 
-#Clean sex for age pyramid graphic
-linelist_cleaned$sex[linelist_cleaned$sex == "U"] <- NA
-linelist_cleaned$sex <- factor(linelist_cleaned$sex)
-
-#Create age groups for pyramid graphic
-linelist_cleaned$age_group <- age_categories(linelist_cleaned$age_years, 
-                                             breakers = c(0, 3, 15, 30, 45))
 ```
 
 
-## The Here packge and here() function  
+### Factors and the forcats() package  
 
-The templates use the **Here package to assist with locating local files**. The Here package and its function here() work well with R projects.  
+#### Factors
 
-**How it works:** When the package "Here" is loaded, it automatically considers the top-level folder of your R project as "here" - as a benchmark for all other files in the Project. In your script, if you want to import or reference a file saved in your R project's folders, you use the function here() to tell R where the file is in relation to "here".  
+Factors are a special variable class in which data are stored internally as integers, but each integer *level* has a corresponding character value *label* that is used when the data are displayed. This can be partularly helpful when running statistical models which expect numeric (e.g. 0/1) inputs.  
 
-If you are unsure where "here" is set to, run this command:
+*Factors are similar to labeled values of categorical variables in Stata.*  
 
-```{r eval=FALSE}
-# This command tells you the folder path that "here" is set to 
-here()
-```
+You can read more about factors [here](https://www.stat.berkeley.edu/~s133/factors.html).  
```

Comment:
It's better to link to a descriptive phrase as opposed to simply the word "here" because it's much more accessible for screen readers (otherwise, the user gets a bunch of links that say "here" with random URLs) https://www.w3.org/QA/Tips/noClickHere

```suggestion
You can [read more about factors here](https://www.stat.berkeley.edu/~s133/factors.html).  
```